internal/colonyapi/colonyapi_test.go

```diff
@@ -0,0 +1,93 @@
+package colonyapi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestAPI_ValidateApiKey(t *testing.T) {
+	ctx := context.TODO()
+	validToken := "super-duper-valid-token"
+	apiEndpoint := "/api/v1/token/validate"
+
+	t.Run("valid api key", func(t *testing.T) {
+		mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			if r.URL.Path != apiEndpoint {
+				t.Fatalf("expected to request %s but got: %s", apiEndpoint, r.URL.Path)
+			}
+
+			// check Authorization header is set properly
+			if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", validToken) {
+				w.WriteHeader(http.StatusUnauthorized)
+				w.Write([]byte("Authentication not provided"))
+			}
+
+			w.Write([]byte("{\"isValid\": true}"))
+		}))
+
+		defer mockServer.Close()
+
+		api := New(mockServer.URL, validToken)
+
+		err := api.ValidateApiKey(ctx)
+		if err != nil {
+			t.Fatalf("expected nil but got: %s", err)
+		}
+	})
+
+	t.Run("invalid api key", func(t *testing.T) {
+		mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("{\"isValid\": false}"))
```

Comment:
Here's a trick for you. By now, you see that any `.Write()` function takes a byte array that, as you saw, you had to escape to appropriately print it. Wouldn't it be nice if there was a different way?

One way could be creating a variable then just casting it, avoiding the quote-escaping altogether:

```suggestion
			resp := `{"isValid": false}`
			w.Write([]byte(resp))
```

But that's IMHO too long... What we can do though is use another piece from the standard library:

```suggestion
			fmt.Fprintln(w, `{"isValid": false}`)
```

This line becomes as readable as you can think of. Since `w` has a method called `.Write()`, Go's `fmt.Fprint*` can receive as a first parameter "anything you can write to" (and `http.ResponseWriter` is something you can write to, since it has the `.Write()` method) so `fmt` works with it too. Wanna print something to `stderr`?

```go
fmt.Fprintln(os.Stderr, "this, in a program, is printed to bash' stderr!")
```